File: src/nomanssave/gZ.java
package nomanssave;

import java.io.FilterOutputStream;
import java.io.OutputStream;
import net.jpountz.lz4.LZ4Compressor;
import net.jpountz.lz4.LZ4Factory;

public class gZ extends FilterOutputStream {
  private static final LZ4Factory sb = LZ4Factory.safeInstance();
  
  private final LZ4Compressor sc = sb.fastCompressor();
  
  private byte[] buffer = new byte[524288];
  
  private int sd = 0;
  
  private int se = 0;
  
  private int sf = 0;
  
  public gZ(OutputStream paramOutputStream) {
    super(paramOutputStream);
  }
  
  public void write(int paramInt) {
    if (this.sd == this.buffer.length)
      ek(); 
    this.buffer[this.sd++] = (byte)paramInt;
  }
  
  public void write(byte[] paramArrayOfbyte) {
    write(paramArrayOfbyte, 0, paramArrayOfbyte.length);
  }
  
  public void write(byte[] paramArrayOfbyte, int paramInt1, int paramInt2) {
    if (paramInt2 == this.buffer.length)
      ek(); 
    while (paramInt2 >= this.buffer.length - this.sd) {
      int i = this.buffer.length - this.sd;
      System.arraycopy(paramArrayOfbyte, paramInt1, this.buffer, this.sd, i);
      this.sd = this.buffer.length;
      ek();
      paramInt1 += i;
      paramInt2 -= i;
    } 
    if (paramInt2 > 0) {
      System.arraycopy(paramArrayOfbyte, paramInt1, this.buffer, this.sd, paramInt2);
      this.sd += paramInt2;
    } 
  }
  
  private void ek() {
    int i = this.sc.maxCompressedLength(this.sd);
    byte[] arrayOfByte1 = new byte[i];
    int j = this.sc.compress(this.buffer, 0, this.sd, arrayOfByte1, 0, i);
    byte[] arrayOfByte2 = new byte[16];
    arrayOfByte2[0] = -27;
    arrayOfByte2[1] = -95;
    arrayOfByte2[2] = -19;
    arrayOfByte2[3] = -2;
    arrayOfByte2[4] = (byte)(0xFF & j);
    arrayOfByte2[5] = (byte)(0xFF & j >> 8);
    arrayOfByte2[6] = (byte)(0xFF & j >> 16);
    arrayOfByte2[7] = (byte)(0xFF & j >> 24);
    arrayOfByte2[8] = (byte)(0xFF & this.sd);
    arrayOfByte2[9] = (byte)(0xFF & this.sd >> 8);
    arrayOfByte2[10] = (byte)(0xFF & this.sd >> 16);
    arrayOfByte2[11] = (byte)(0xFF & this.sd >> 24);
    this.out.write(arrayOfByte2);
    this.out.write(arrayOfByte1, 0, j);
    this.se += this.sd;
    this.sd = 0;
    this.sf += j + 16;
  }
  
  public int ch() {
    return this.se;
  }
  
  public int ci() {
    return this.sf;
  }
  
  public void flush() {
    if (this.sd > 0)
      ek(); 
    this.out.flush();
  }
  
  public void close() {
    try {
      if (this.sd > 0)
        ek(); 
    } finally {
      this.out.close();
    } 
  }
}


/* Location:              E:\NMSSE_Source\NMSSaveEditor.jar!\nomanssave\gZ.class
 * Java compiler version: 8 (52.0)
 * JD-Core Version:       1.1.3
 */

File: NMSSaveEditor/nomanssave/mixed/gZ.cs
using K4os.Compression.LZ4;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

namespace NMSSaveEditor
{

public class gZ : FilterOutputStream {
   public static LZ4Factory se = null /*LZ4Factory*/;
   public LZ4Compressor sf;
   public byte[] buffer;
   public int sg;
   public int sh;
   public int si;

   public gZ(Stream var1) {
      base(var1);
      this.sf = se.fastCompressor();
      this.buffer = new byte[524288];
      this.sg = 0;
      this.sh = 0;
      this.si = 0;
   }

   public void write(int var1) {
      if (this.sg == this.buffer.Length) {
         this.ek();
      }

      this.buffer[this.sg++] = (byte)var1;
   }

   public void write(byte[] var1) {
      this.Write(var1, 0, var1.Length);
   }

   public void write(byte[] var1, int var2, int var3) {
      if (var3 == this.buffer.Length) {
         this.ek();
      }

      while(var3 >= this.buffer.Length - this.sg) {
         int var4 = this.buffer.Length - this.sg;
         Array.Copy(var1, var2, this.buffer, this.sg, var4);
         this.sg = this.buffer.Length;
         this.ek();
         var2 += var4;
         var3 -= var4;
      }

      if (var3 > 0) {
         Array.Copy(var1, var2, this.buffer, this.sg, var3);
         this.sg += var3;
      }

   }

   public void ek() {
      int var1 = this.sf.maxCompressedLength(this.sg);
      byte[] var2 = new byte[var1];
      int var3 = this.sf.compress((byte[])this.buffer, 0, this.sg, (byte[])var2, 0, var1);
      byte[] var4 = new byte[]{229, 161, 237, 254, (byte)(255 & var3), (byte)(255 & var3 >> 8), (byte)(255 & var3 >> 16), (byte)(255 & var3 >> 24), (byte)(255 & this.sg), (byte)(255 & this.sg >> 8), (byte)(255 & this.sg >> 16), (byte)(255 & this.sg >> 24), 0, 0, 0, 0};
      this.@out.Write(var4);
      this.@out.Write(var2, 0, var3);
      this.sh += this.sg;
      this.sg = 0;
      this.si += var3 + 16;
   }

   public int ch() {
      return this.sh;
   }

   public int ci() {
      return this.si;
   }

   public void flush() {
      if (this.sg > 0) {
         this.ek();
      }

      this.@out.Flush();
   }

   public void close() {
      try {
         if (this.sg > 0) {
            this.ek();
         }
      } finally {
         this.@out.Close();
      }

   }
}

}
